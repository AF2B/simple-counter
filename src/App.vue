<!-- <script setup lang="ts">
  import { type Ref, ref, Ref } from 'vue';

  const counter: Ref<number> = ref<number>(0);

  const increment: () => void = (): void => {
    counter.value += 1;
  };

  const decrement: () => void = (): void => {
    counter.value -= 1;
  };
</script> Default mode-->

<!-- <script lang="ts">
export default {
  data() {
    return {
      counter: 0,
    };
  },
  methods: {
    increment() {
      this.counter++;
    },
    decrement() {
      this.counter--;
    },
  },
};
</script> Option API-->

<!-- Composition API -->
<!-- <script lang="ts">
import { ref } from 'vue';

export default {
  setup() {
    const counter = ref(0);

    const increment = () => {
      counter.value++;
    };

    const decrement = () => {
      counter.value--;
    };

    return {
      counter,
      increment,
      decrement,
    };
  },
};
</script> -->

<!-- Composition API with <script setup> -->
<script lang="ts" setup>
  import { ref, type Ref } from 'vue';

  const counter: Ref<number> = ref<number>(0);

  const increment: () => void = (): void => {
    counter.value += 1;
  };

  const decrement: () => void = (): void => {
    counter.value -= 1;
  };
</script>

<template>
  <main>
    <div>
      <h1>Just an counter with Vue 3</h1>
    </div>
    <div>
      <h2>Counter: {{ counter }}</h2>
      <button @click="decrement">Decrement</button>
      <button @click="increment">Increment</button>
    </div>
  </main>
</template>

<style scoped>
  main {
    background-color: aliceblue;
    width: 100vw;
    height: 100vh;
    display: flex;
    justify-content: center;
    align-items: center;
    text-align: center;
    flex-direction: column;
  }

  main > div {
    margin: 1rem;
  }
</style>
