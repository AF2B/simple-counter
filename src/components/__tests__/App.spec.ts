import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import App from '../../App.vue'

describe('App', () => {
  it('renders correctly', () => {
    const result = mount(App).html()
    expect(result).toMatchSnapshot()
  })

  it('renders the initial counter value', () => {
    const wrapper = mount(App)
    const counterText = wrapper.find('h2').text()
    expect(counterText).toBe('Counter: 0')
  })

  it('increments the counter on button click', async () => {
    const wrapper = mount(App)
    const incrementButton = wrapper.find('button:nth-child(1)')
    await incrementButton.trigger('click')
    const counterText = wrapper.find('h2').text()
    expect(counterText).toBe('Counter: 1')
  })

  it('decrements the counter on button click', async () => {
    const wrapper = mount(App)
    const decrementButton = wrapper.find('button:nth-child(2)')
    await decrementButton.trigger('click')
    const counterText = wrapper.find('h2').text()
    expect(counterText).toBe('Counter: -1')
  })
})
