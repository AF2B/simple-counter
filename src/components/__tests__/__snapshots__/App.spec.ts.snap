// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > renders correctly 1`] = `
"<main data-v-7a7a37b1=\\"\\">
  <div data-v-7a7a37b1=\\"\\">
    <h1 data-v-7a7a37b1=\\"\\">Just an counter with Vue 3</h1>
  </div>
  <div data-v-7a7a37b1=\\"\\">
    <h2 data-v-7a7a37b1=\\"\\">Counter: 0</h2><button data-v-7a7a37b1=\\"\\">Decrement</button><button data-v-7a7a37b1=\\"\\">Increment</button>
  </div>
</main>"
`;
